contracts/savings/SavingsManager.sol

```diff
@@ -0,0 +1,432 @@
+     */
+    function setSavingsRate(uint256 _savingsRate) external onlyGovernor {
+        // Greater than 60% upto 100%
+        require(_savingsRate >= 6e17 && _savingsRate <= 1e18, "Must be a valid rate");
+        savingsRate = _savingsRate;
+        emit SavingsRateChanged(_savingsRate);
+    }
+
+    /**
+     * @dev Allows the liquidator to deposit proceeds from iquidated gov tokens.
+     * Transfers proceeds on a second by second basis to the Savings Contract over 1 week.
+     * @param _mAsset The mAsset to transfer and distribute
+     * @param _liquidated Units of mAsset to distribute
+     */
+    function depositLiquidation(address _mAsset, uint256 _liquidated)
+        external
+        override
+        whenNotPaused
+        onlyLiquidator
+        whenStreamsNotFrozen
+    {
+        // Collect existing interest to ensure everything is up to date
+        _collectAndDistributeInterest(_mAsset);
+
+        // transfer liquidated mUSD to here
+        IERC20(_mAsset).safeTransferFrom(_liquidator(), address(this), _liquidated);
+
+        uint256 leftover = _unstreamedRewards(_mAsset, StreamType.liquidator);
+        _initialiseStream(_mAsset, StreamType.liquidator, _liquidated + leftover, DURATION);
+
+        emit LiquidatorDeposited(_mAsset, _liquidated);
+    }
+
+    /**
+     * @dev Collects the platform interest from a given mAsset and then adds capital to the
+     * stream. If there is > 24h left in current stream, just top it up, otherwise reset.
+     * @param _mAsset The mAsset to fetch interest
+     */
+    function collectAndStreamInterest(address _mAsset)
+        external
+        override
+        whenNotPaused
+        whenStreamsNotFrozen
+    {
+        // Collect existing interest to ensure everything is up to date
+        _collectAndDistributeInterest(_mAsset);
+
+        uint256 currentTime = block.timestamp;
+        uint256 previousBatch = lastBatchCollected[_mAsset];
+        uint256 timeSincePreviousBatch = currentTime - previousBatch;
+        require(timeSincePreviousBatch > 6 hours, "Cannot deposit twice in 6 hours");
+        lastBatchCollected[_mAsset] = currentTime;
+
+        // Batch collect
+        (uint256 interestCollected, uint256 totalSupply) =
+            IMasset(_mAsset).collectPlatformInterest();
+
+        if (interestCollected > 0) {
+            // Validate APY
+            uint256 apy =
+                YieldValidator.validateCollection(
+                    totalSupply,
+                    interestCollected,
+                    timeSincePreviousBatch
+                );
+
+            // Get remaining rewards
+            uint256 leftover = _unstreamedRewards(_mAsset, StreamType.yield);
+            _initialiseStream(_mAsset, StreamType.yield, interestCollected + leftover, ONE_DAY);
+
+            emit InterestCollected(_mAsset, interestCollected, totalSupply, apy);
+        }
+
+        emit InterestCollected(_mAsset, interestCollected, totalSupply, 0);
```

Comment:
I think this needs to be in an `else` from `if (interestCollected > 0)`. At the moment the interest collection transactions are firing two `InterestCollected` events